projects/aws/image-builder/README.md

```diff
@@ -0,0 +1,155 @@
+
+Supported Providers
+* vsphere
+* baremetal
+
+Supported OSes
+* ubuntu
+
+Supported Release Channels
+* 1-23
+* 1-22
+* 1-21
+* 1-20
+
+### Building Node Images for vSphere
+
+Vsphere is one of the supported infrastructure providers the Image builder tool can build EKS-A node images for. In order
+to build a node image for vSphere, the image builder tool needs to run from an environment with network access to vcenter.
+
+#### Pre-reqs for vSphere Image Builder
+* Machine running Ubuntu 22.04 that has access to the vcenter environment and vcenter url. This machine could also run as a vm on vSphere.
+* vSphere user with permissions listed below
+* Minimum machine resources required
+  * 50 GB disk space on the machine
+  * 2 vCPUs
+  * 8 GB RAM
+* Machine as well as vSphere network used below must have network access to
+  * vCenter endpoint
+  * public.ecr.aws (to download container images from EKS-A)
+  * anywhere-assets.eks.amazonaws.com (to download EKS-A binaries)
+  * distro.eks.amazonaws.com (to download EKS-D binaries)
+  * d2glxqk2uabbnd.cloudfront.net (for EKS-A and EKS-D ECR container images)
+
+##### vSphere User Permissions
+Inventory:
+* Create new
+
+Configuration:
+* Change configuration
+* Add new disk
+* Add or remove device
+* Change memory
+* Change settings
+* Set annotation
+
+Interaction:
+* Power on
+* Power off
+* Console interaction
+* Configure CD media
+* Device connection
+
+Snapshot management:
+* Create snapshot
+
+Provisioning
+
+* Mark as template
+
+Resource Pool
+* Assign vm to resource pool
+
+Datastore
+* Allocate space
+* Browse data
+* Low level file operations
+
+Network
+* Assign network to vm
+
+#### Building a vSphere OVA Node Image
+1. Install pre-requisite packages
+```
+sudo apt update -y
+sudo apt install jq unzip make ansible -y
+sudo snap install yq
+```
+2. Build or download the image builder tool
+3. Create a content library on vSphere
+```
+govc library.create "<library name>"
+```
+4. Create vsphere-connection.json config file
+```
+{
+  "cluster":"<vsphere cluster used for image building>",
+  "convert_to_template":"false",
+  "create_snapshot":"<creates a snapshot on base OVA after building if set to true>",
+  "datacenter":"<vsphere datacenter used for image building>",
+  "datastore":"<datastore used to store template/for image building>",
+  "folder":"<folder on vsphere to create temporary vm>",
+  "insecure_connection":"true",
+  "linked_clone":"false",
+  "network":"<vsphere network used for image building>",
+  "password":"<vcenter username>",
+  "resource_pool":"<resource pool used for image building vm>",
+  "username":"<vcenter username>",
+  "vcenter_server":"<vcenter fqdn>",
+  "vsphere_library_name": "<vsphere content library name>"
+}
+```
+5. Run the image builder tool for appropriate release channel
+```
+image-builder build --os ubuntu --hypervisor vsphere --vsphere-config <path to above json file> --release-channel <release channel>
```

Comment:
release channel being `1-20` for ex? i know you defined them above but i was confused at first read, not sure if there is a way to spell it out better here?

Reply:
yea I can put in an example here